[bug] failed to build android apk

### Describe the bug

cargo, rust,npm,tauri-cli are all latest.
use `npx tauri android build` to build.
the wrong message is below:
w: d:\pf\rust\.cargo\registry\src\mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd\tauri-2.0.0-alpha.9\mobile\android\src\m
ain\java\app\tauri\PermissionHelper.kt: (113, 30): Unchecked cast: Array<String> to Array<String?>
w: d:\pf\rust\.cargo\registry\src\mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd\tauri-2.0.0-alpha.9\mobile\android\src\m
ain\java\app\tauri\plugin\PluginMethodData.kt: (11, 23): Parameter 'methodDecorator' is never used
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:rustBuildArm64Release'.
> A problem occurred starting process 'command 'npx-cli.cmd''

### Reproduction

_No response_

### Expected behavior

_No response_

### Platform and versions

```text
Microsoft Windows [版本 10.0.19045.3086]
(c) Microsoft Corporation。保留所有权利。

D:\workspace\tauri-apps\na>npx tauri info

[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 114.0.1823.43
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.72.0-nightly (7820972f8 2023-06-10)
    ✔ Cargo: 1.72.0-nightly (49b6d9e17 2023-06-09)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: nightly-x86_64-pc-windows-msvc (default)
    - node: 20.0.0
    - pnpm: 8.6.2
    - yarn: 1.22.19
    - npm: 9.7.1

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.9
    - tauri-build [RUST]: 2.0.0-alpha.5
    - wry [RUST]: 0.28.3
    - tao [RUST]: 0.19.1
    - @tauri-apps/api [NPM]: 2.0.0-alpha.4
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.9

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ./dist
    - devPath: ./dist
```


### Stack trace

_No response_

### Additional context

_No response_

--- a/.changes/cli-npx-mobile.md
+++ b/.changes/cli-npx-mobile.md
@@ -1,0 +1,6 @@
+---
+'tauri-cli': 'patch'
+'@tauri-apps/cli': 'patch'
+---
+
+Fix `tauri (android|ios) (dev|build)` failing when using `npx tauri`

--- a/.changes/cli-npx-mobile.md
+++ b/.changes/cli-npx-mobile.md
@@ -1,6 +1,6 @@
 ---
-'tauri-cli': 'patch'
-'@tauri-apps/cli': 'patch'
+'tauri-cli': 'patch:bug'
+'@tauri-apps/cli': 'patch:bug'
 ---
 
 Fix `tauri (android|ios) (dev|build)` failing when using `npx tauri`

--- a/tooling/cli/node/index.js
+++ b/tooling/cli/node/index.js
@@ -21,7 +21,7 @@ function isMusl() {
   // For Node 10
   if (!process.report || typeof process.report.getReport !== 'function') {
     try {
-      const lddPath = require('child_process').execSync('which ldd').toString().trim();
+      const lddPath = require('child_process').execSync('which ldd').toString().trim()
       return readFileSync(lddPath, 'utf8').includes('musl')
     } catch (e) {
       return true

--- a/tooling/cli/src/build.rs
+++ b/tooling/cli/src/build.rs
@@ -269,7 +269,7 @@ pub fn setup(options: &mut Options, mobile: bool) -> Result<AppInterface> {
     .any(|ch| !(ch.is_alphanumeric() || ch == '-' || ch == '.'))
   {
     error!(
-      "The bundle identifier \"{}\" set in `{} > tauri > bundle > identifier`. The bundle identifier string must contain only alphanumeric characters (A–Z, a–z, and 0–9), hyphens (-), and periods (.).",
+      "The bundle identifier \"{}\" set in `{} > tauri > bundle > identifier`. The bundle identifier string must contain only alphanumeric characters (A-Z, a-z, and 0-9), hyphens (-), and periods (.).",
       config_.tauri.bundle.identifier,
       bundle_identifier_source
     );

--- a/tooling/cli/src/mobile/init.rs
+++ b/tooling/cli/src/mobile/init.rs
@@ -139,7 +139,7 @@ pub fn exec(
       } else {
         manager_stem
       };
-      if !build_args.is_empty() && !is_npx {
+      if !(build_args.is_empty() || is_npx) {
         // remove script path, we'll use `npm_lifecycle_event` instead
         build_args.remove(0);
       }

--- a/tooling/cli/src/mobile/init.rs
+++ b/tooling/cli/src/mobile/init.rs
@@ -130,20 +130,26 @@ pub fn exec(
   if r.is_match(&bin_stem) {
     if let Some(npm_execpath) = var_os("npm_execpath").map(PathBuf::from) {
       let manager_stem = npm_execpath.file_stem().unwrap().to_os_string();
-      binary = if manager_stem == "npm-cli" {
+      let is_npm = manager_stem == "npm-cli";
+      let is_npx = manager_stem == "npx-cli";
+      binary = if is_npm {
         "npm".into()
+      } else if is_npx {
+        "npx".into()
       } else {
         manager_stem
       };
-      if !build_args.is_empty() {
+      if !build_args.is_empty() && !is_npx {
         // remove script path, we'll use `npm_lifecycle_event` instead
         build_args.remove(0);
       }
-      if binary == "npm" {
+      if is_npm {
         build_args.insert(0, "--".into());
       }
-      build_args.insert(0, var("npm_lifecycle_event").unwrap());
-      if binary == "npm" {
+      if !is_npx {
+        build_args.insert(0, var("npm_lifecycle_event").unwrap());
+      }
+      if is_npm {
         build_args.insert(0, "run".into());
       }
     }

